### PR TITLE
List selection event support

### PR DESCRIPTION
**Pull request by Petikoch**
Best regards,
Peti

### CODE REVIEW

**Comment by Petikoch** [comment]
Hi everyone,
Hi @eddieburns55, 

would be nice to get this in and bake a new RxSwing release... I'm waiting for the release of the recently merged WindowEventSource, too.

Best regards,
Peti

**Comment by epb-644** [comment]
Hi Peti, thanks for the PR. Sorry, I started a new job recently so it may take me a few days to review it. I'll try to cut a new release once we merge this ListSelection support in.

**Comment by Petikoch** [comment]
Hi @eddieburns55 ,

thanks for your feedback. 

I wish you a good start in the new job!

Best regards from Switzerland,
Peti

File: src/test/java/rx/swing/sources/ListSelectionEventSourceTest.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+import javax.swing.*;
+import javax.swing.event.ListSelectionEvent;
+
+import static junit.framework.Assert.assertEquals;
+
+public class ListSelectionEventSourceTest {
+
+    @Test
+    public void jtableRowSelection_observingSelectionEvents() throws Throwable {
```
Comment (epb-644) [inline src/test/java/rx/swing/sources/ListSelectionEventSourceTest.java]: Use camel-cased method names for consistency (jtableRowSelectionObservingSelectionEvents)

File: src/test/java/rx/swing/sources/ListSelectionEventSourceTest.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+                table.getSelectionModel().setSelectionInterval(0, 0);
+
+                testSubscriber.assertNoErrors();
+                testSubscriber.assertValueCount(1);
+
+                assertListSelectionEventEquals(
+                        new ListSelectionEvent(
+                                table.getSelectionModel(),
+                                0 /* start of region with selection changes */,
```
Comment (epb-644) [inline src/test/java/rx/swing/sources/ListSelectionEventSourceTest.java:54]: Thanks for the comments here. We should add more descriptions like this to the rest of our tests.

File: src/test/java/rx/swing/sources/ListSelectionEventSourceTest.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
+/**
```
Comment (epb-644) [inline src/test/java/rx/swing/sources/ListSelectionEventSourceTest.java:1]: In general, let's file separate PRs for general cleanup. That said, thanks for spotting these, and I'm ok with including them this time.

File: src/test/java/rx/swing/sources/ListSelectionEventSourceTest.java [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+                                0 /* start of region with selection changes */,
+                                2 /* end of region with selection changes */,
+                                false),
+                        testSubscriber.getOnNextEvents().get(1));
+            }
+        }).awaitTerminal();
+    }
+
+    @Test
+    public void jtableRowSelection_unsubscribe_removesRowSelectionListener() throws Throwable {
```
Comment (epb-644) [inline src/test/java/rx/swing/sources/ListSelectionEventSourceTest.java]: :thumbsup:

**Comment by epb-644** [comment]
Please remove the underscores in method names in ListSelectionEventSourceTest. Otherwise, everything looks good.

**Comment by Petikoch** [comment]
Thanks @eddieburns55 for the review. I updated the pull request.

Best regards,
Peti

**Comment by Petikoch** [comment]
@eddieburns55, when do you think the new release will be available? Best regards,Peti

**Comment by epb-644** [comment]
Hi @Petikoch, I'll work on it this weekend. Thanks for reminding me. :)

**Pull request merged**